pkg/osquery/tables/execparsers/remotectl/parse.go

```diff
@@ -0,0 +1,128 @@
+//go:build darwin
+// +build darwin
+
+package remotectl
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+)
+
+// parseDumpstate parses the result of /usr/libexec/remotectl into a map that can be flattened by dataflatten.
+// We expect results in the following format, with empty newlines separating devices:
+//
+//	<device name>
+//			Key: Value
+//			Properties: {
+//				Key => Value
+//			}
+//			Services:
+//				service1
+//				service2
+func parseDumpstate(reader io.Reader) (any, error) {
+	results := make(map[string]map[string]interface{})
+
+	currentDeviceName := ""
+	insidePropertiesDictionary := false
+	insideServicesArray := false
+
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// We need a device name to proceed -- if present, initialize its data in `results` and move on
+		if isDeviceName(line) {
+			currentDeviceName = extractDeviceName(line)
+			results[currentDeviceName] = make(map[string]interface{})
+			results[currentDeviceName]["Properties"] = make(map[string]interface{})
+			results[currentDeviceName]["Services"] = make([]string, 0)
+			continue
+		}
+
+		if currentDeviceName == "" {
+			return nil, fmt.Errorf("no device name(s) given in remotectl dumpstate output")
+		}
+
+		if isDeviceDelimiter(line) {
+			// We've reached a new device -- reset state and continue
+			currentDeviceName = ""
+			insidePropertiesDictionary = false
+			insideServicesArray = false
+			continue
+		}
+
+		line = strings.TrimSpace(line)
+
+		if insidePropertiesDictionary {
+			if line == "}" {
+				insidePropertiesDictionary = false
+				continue
+			}
+
+			propertyKey, propertyValue, err := extractPropertyKeyValue(line)
+			if err != nil {
+				return nil, err
+			}
+
+			properties := results[currentDeviceName]["Properties"].(map[string]interface{})
+			properties[propertyKey] = propertyValue
+		} else if insideServicesArray {
```

Comment:
I think we should use `continue` instead of `else` or `else if` when possible through out the code. I believe it makes the code easier to read by reducing nesting.

For example:
```suggestion
			properties[propertyKey] = propertyValue
			continue
		if insideServicesArray {
```

Reply:
Agreed that is more readable -- I'll update to make this change throughout 🙂

Reply:
This should be all set now

Reply:
yeah, broad support for easy returns over complex if/else code